lib/engine/step/g_1846/train.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+require_relative '../train'
+
+module Engine
+  module Step
+    module G1846
+      class Train < Train
+        def setup
+          super
+          @issued = false
+        end
+
+        def actions(entity)
+          return [] if entity.minor? || entity.company?
+
+          return [] if entity.owner == @game.share_pool
+
+          if entity.corporation? && must_buy_train?(entity)
```

Comment:
if must_buy_train

Reply:
`entity` can be the corp's president here, must_buy_train? blows up if a player is passed

53519559